src/v/cluster/scheduling/types.h

```diff
@@ -360,4 +360,25 @@ struct allocation_request {
     friend std::ostream& operator<<(std::ostream&, const allocation_request&);
 };
 
+struct raw_allocation_request {
```

Comment:
Can we please add comments to both of the request structures and mention why one should use one or the other? So someone doesn't try to "cleanup" the code later and they miss the memory blowup.